### ISSUE TITLE
Help - can't pass the login page

### ISSUE THREAD

**Issue by ranrubin** [allegroai/trains-server#50]
Hi, I would really appreciate any help :)

I had a running trains server that was deployed using docker-compose on a linux machine.
It worked nicely just the other day and suddenly I can't login into the application (can't pass the type and START). 
I tried to reinstall it, but nothing helps. Any Idea where the issue can be? 
I can't find anything in the  apiserver.log .

**Comment by jkhenning** [comment]
Hi @ranrubin ,

Can you please elaborate on what do you mean by "can't pass the type and START"? Can you perhaps attach a screenshot?
Also, if we're talking about the web-browser, can you please open the developer's panel (usually using F12), go to the **Network** panel and reload the page, than take another screenshot (or share what you see there)?

**Comment by ranrubin** [comment]
Sorry for the late response 

I suspect that there is an issue with my infrastructure. I will appreciate any hints regarding where can I look for a solution / what will help me debug :)

![trains](https://user-images.githubusercontent.com/27898148/88061550-50c3b400-cb70-11ea-943c-0e1e8aad60c5.png)

**Comment by orgoro** [comment]
I have the same,
logging from certain ip works from other I get the same error
![image](https://user-images.githubusercontent.com/20637412/88519194-64509e00-cffa-11ea-9310-3feabf2a70ea.png)

**Comment by jkhenning** [comment]
Hi @orgoro,

Is there any consistent difference between the IPs you can access from, and those from which you can't? Might they be routed differently, or is there any Firewall or Load Balancer with different rules for them? It looks as if the calls were simply blocked along the way...

**Comment by orgoro** [comment]
Hi @jkhenning,
- Nothing special with the IP - maybe only the fact that it was used for initial login
- Since then I moved the server to fixed users login
- The firewall is configured to accept all incoming on the relevant ports
- I manage to see the website but instead of the user + password form I see the screen like @ranrubin 
![image](https://user-images.githubusercontent.com/20637412/88568204-422e3e80-d041-11ea-9046-26c9b338a835.png)

**Comment by jkhenning** [comment]
@orgoro can you run a Trains example with this server? Even just a few lines of code that do `Task.init()`?

**Comment by orgoro** [comment]
@jkhenning yes everything works fine also when I connect from a different IP the web server works.
Only trouble logging in from a certain IP. 
Cleared cache, hard reset nothing worked. 
Also tried resetting the server

**Comment by jkhenning** [comment]
@orgoro are you using port `8080` to access the web server?

**Comment by orgoro** [comment]
@jkhenning yes

**Comment by jkhenning** [comment]
@orgoro using http? https? I'm thinking the difference might be an unnoticed auto-complete in the browser that uses different ports, causing the pre-loaded WebApp to try and communicate with the server on an incorrect port...

**Comment by orgoro** [comment]
@jkhenning http, port 8080
![image](https://user-images.githubusercontent.com/20637412/88656943-c46a4180-d0d9-11ea-80e8-c6b649b84d44.png)
![image](https://user-images.githubusercontent.com/20637412/88656977-d4822100-d0d9-11ea-9d8d-8d45c833f073.png)

**Comment by jkhenning** [comment]
Hi @orgoro ,

Can you please send the full details for some (or all) of the failed requests? You can Right-click any of the requests and choose Copy -> "Copy all as cURL" (to get all the requests in the page with all headers) as well as choose a few and do Right-click than Copy -> "Copy response headers" to get the response headers.

Also, following a different (and equally strange 😄) [issue](#55), can you try browsing to the web-app in incognito/private mode?

**Comment by ranrubin** [comment]
I'll add to the mystery by saying that it doesn't work from my office but does work from home/hotspot (attempted from the same computer and browser). Also, in order to debug, I had the instance accept **all** incoming traffic. 

I thought it might be that the office's internet provider blocks automatically some CIDR blocks,  but then why would I be able to see the homepage...

**Comment by jkhenning** [comment]
@ranrubin this sounds like there must be a Firewall in your office that blocks these calls... When you say HotSpot I assume you mean using your smartphone with the Cellular Provider's network (and not the office network)?

**Comment by ranrubin** [comment]
@jkhenning  Exactly. But I kind of dismissed this possibility because I can see the homepage. 

If the trains' ip is 1.2.3.4. and I manage to get to the homepage at 1.2.3.4:8080, from the office internet, what is blocked? 
Obviously not 1.2.3.4 :)

**Comment by jkhenning** [comment]
Well, `8080` isn't usually blocked, but It could be port `8008` that is blocked (this is the port used to access the apiserver). Can you try testing that using `http://<address>:8008`? You should get a response if it works (even a response that says there's an error :), and a timeout if it's blocked...

**Comment by ranrubin** [comment]
Tried with `telnet` and managed to connect.

**Comment by jkhenning** [comment]
@ranrubin Telnet may work while browser will not. This can happen if the firewall inspects the traffic, and sees the method, or specific information transferred.

**Comment by orgoro** [comment]
@jkhenning incognito not working as well

**Comment by jkhenning** [comment]
@orgoro can you please try testing the same thing:
>Can you try testing that using `http://<address>:8008`? You should get a response if it works (even a response that says there's an error :), and a timeout if it's blocked...
